src/js/controller/instream-adapter.js

```diff
@@ -208,14 +208,16 @@ var InstreamAdapter = function(_controller, _model, _view) {
 
         _this.addClickHandler();
 
+        adModel.set('skipButton', false);
+
+        const playPromise = _instream.load(item);
```

Comment:
This variable seems unnecessary

Reply:
We need to return the play promise we get from ```_instream.load```

Reply:
You might be right. Seems like we do not use the promise anywhere.

Reply:
You should use it to capture ad start failures in vast.

Reply:
And to only fire ad impression on `then`